ios/browser/api/url/url_pattern_ios.h

```diff
@@ -0,0 +1,119 @@
+  URLPatternIOSSchemeUUIDInPackage = 1 << 11,
+  URLPatternIOSSchemeAll = -1
+} NS_SWIFT_NAME(URLPattern.Scheme);
+
+typedef NSInteger URLPatternIOSParseResult NS_TYPED_ENUM
+    NS_SWIFT_NAME(URLPattern.ParseResult);
+OBJC_EXPORT URLPatternIOSParseResult const URLPatternIOSParseResultSuccess;
+OBJC_EXPORT URLPatternIOSParseResult const
+    URLPatternIOSParseResultMissingSchemeSeparator;
+OBJC_EXPORT URLPatternIOSParseResult const
+    URLPatternIOSParseResultInvalidScheme;
+OBJC_EXPORT URLPatternIOSParseResult const
+    URLPatternIOSParseResultWrongSchemeSeparator;
+OBJC_EXPORT URLPatternIOSParseResult const URLPatternIOSParseResultEmptyHost;
+OBJC_EXPORT URLPatternIOSParseResult const
+    URLPatternIOSParseResultInvalidHostWildcard;
+OBJC_EXPORT URLPatternIOSParseResult const URLPatternIOSParseResultEmptyPath;
+OBJC_EXPORT URLPatternIOSParseResult const URLPatternIOSParseResultInvalidPort;
+OBJC_EXPORT URLPatternIOSParseResult const URLPatternIOSParseResultInvalidHost;
+
+/// A wrapper around Chromium's `URLPattern` functionality
+OBJC_EXPORT
+NS_SWIFT_NAME(URLPattern)
+@interface URLPatternIOS : NSObject <NSCopying>
+
+/// A pattern that will match all urls.
+///
+/// Can be passed into `parsePattern` or `initWithValidSchemes:patternLiteral:`
+@property(class, readonly) NSString* allURLsPattern;
+
+@property(nonatomic) URLPatternIOSScheme validSchemes;
+@property(nonatomic) NSString* scheme;
+@property(nonatomic) NSString* host;
+@property(nonatomic) NSString* port;
+@property(nonatomic) NSString* path;
+@property(nonatomic) bool isMatchingSubdomains;
+@property(nonatomic) bool isMatchingAllURLs;
+
+/// Convenience to construct an empty URLPattern with no schemes setup
+- (instancetype)init;
+
+- (instancetype)initWithValidSchemes:(URLPatternIOSScheme)schemes
+    NS_DESIGNATED_INITIALIZER;
+
+/// Convenience to construct a URLPattern from a string literal. If the string
+/// is not known ahead of time, use `parsePattern:` instead
+- (instancetype)initWithValidSchemes:(URLPatternIOSScheme)schemes
+                      patternLiteral:(NSString*)patternLiteral;
+
+/// Sets the current pattern to match against
+- (URLPatternIOSParseResult)parsePattern:(NSString*)pattern
+    NS_SWIFT_NAME(parse(pattern:));
+
+/// Returns true if the specified scheme can be used in this URL pattern, and
+/// false otherwise.
+- (bool)isValidScheme:(NSString*)scheme;
+
+/// Returns true if this instance matches the specified URL. Always returns
+/// false for invalid URLs.
+- (bool)matchesURL:(NSURL*)url;
+
+/// Returns true if this instance matches the specified security origin.
+- (bool)matchesSecurityOrigin:(NSURL*)origin;
+
+/// Returns true if `scheme` matches our scheme.
+/// Note that if scheme is "filesystem", this may fail whereas `matchesURL`
+/// may succeed.  `matchesURL` is smart enough to look at the inner url instead
+/// of the outer "filesystem:" part.
+- (bool)matchesScheme:(NSString*)scheme;
+
+// Returns true if `host` matches our host.
```

Comment:
`///` instead of `//` (Can be option clicked to see the description that way)

Reply:
Thanks, missed these accidentally!